File: leetcode/algorithms/171_excel_sheet_column_number/main.py
class ExcelSheetColumnNumber:
    """
    # Complexities:
    #   N - Size of `columnTitle`
    #   - Time Complexity: O(N²)
    #   - Space Complexity: O(1)
    """
    def titleToNumber(self, columnTitle: str) -> int:
        result = 0

        for i in range(len(columnTitle)):
            result += 26**i * (ord(columnTitle[::-1][i]) - 64)

        return result


    # Solution
    """
    # Complexities:
    #   N - Size of `columnTitle`
    #   - Time Complexity: O(N²)
    #   - Space Complexity: O(1)
    """
    def solution(self, columnTitle: str) -> int:
        result = 0

        for char in columnTitle:
            val = ord(char) - ord('A') + 1

            result = result * 26 + val

        return result

File: leetcode/algorithms/171_excel_sheet_column_number/main.cpp
#include <string>
using namespace std;

class ExcelSheetColumnNumber {
public:
    /**
     * Complexities:
     *   N - Size of `columnTitle`
     *   - Time Complexity: O(N²)
     *   - Space Complexity: O(1)
     */
    int titleToNumber(string columnTitle) {
        int length = columnTitle.size();
        int result = 0;

        for (int i = 0; i < length; i++) {
            result += ((int)columnTitle[length - i - 1] - 64) * pow(26, i);
        }

        return result;
    }


    // Solution
    /**
     * Complexities:
     *   N - Size of `columnTitle`
     *   - Time Complexity: O(N)
     *   - Space Complexity: O(1)
     */
    int solution(string columnTitle) {
        int result = 0;

        for (char c : columnTitle) {
            int d = c - 'A' + 1;
            result = result * 26 + d;
        }

        return result;
    }
};
